src/parser/demonhunter/havoc/modules/spells/BladeDance.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import SPECS from 'game/SPECS';
+import Events, { DamageEvent, FightEndEvent } from 'parser/core/Events';
+import { ThresholdStyle, When } from 'parser/core/ParseResults';
+import SpellLink from 'common/SpellLink';
+
+//For Blade dance and Death Sweep
+class BladeDance extends Analyzer {
+  get suggestionThresholds() {
+    return {
+      actual: this.badCast,
+      isGreaterThan: {
+        minor: 0,
+        average: 0,
+        major: 1,
+      },
+      style: ThresholdStyle.NUMBER,
+    };
+  }
+
+  badCast = 0;
+  hitCount = 0;
+  hitTimeStamp: number = 0;
+  firstHitTimeStamp: number = 0;
+  strikeTime: number = 1000;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.spec === SPECS.HAVOC_DEMON_HUNTER;
+    if (!this.active || this.selectedCombatant.hasTalent(SPELLS.TRAIL_OF_RUIN_TALENT) || this.selectedCombatant.hasTalent(SPELLS.FIRST_BLOOD_TALENT)) {
+        return;
+    }
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell(SPELLS.DEATH_SWEEP_DAMAGE), this.onDamage);
```

Comment:
If all of these listeners go down to a single method then you could put all the spells in the `.spell` call by putting them in an array like this `.spell([SPELLS.DEATH_SWEEP_DAMAGE, BLADE_DANCE_DAMAGE, ...)`